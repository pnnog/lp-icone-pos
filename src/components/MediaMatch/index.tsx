import styled, { css } from 'styled-components';
import media, { DefaultBreakpoints } from 'styled-media-query';

type breakpoints = keyof DefaultBreakpoints;

type MediaMatchProps = {
  lessThan?: breakpoints;
  greaterThan?: breakpoints;
};

const mediaMatchModifiers = {
  lessThan: (size: breakpoints) => css`
    ${media.lessThan(size)`
      display:block;
    `}
  `,

  greaterThan: (size: breakpoints) => css`
    ${media.greaterThan(size)`
    display:block;
    `}
  `
};
const MediaMatch = styled.div<MediaMatchProps>`
  ${({ lessThan, greaterThan }) => css`
    display: none;
    ${lessThan && mediaMatchModifiers.lessThan(lessThan)}
    ${greaterThan && mediaMatchModifiers.greaterThan(greaterThan)};
  `}
`;

export default MediaMatch;
