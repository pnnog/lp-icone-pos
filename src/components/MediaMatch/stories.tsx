import { Meta, Story } from '@storybook/react';
import MediaMatch from '.';

export default {
  title: 'MediaMatch',
  component: MediaMatch,
  args: {}
} as Meta;

export const Basic: Story = (args) => <MediaMatch {...args} />;
