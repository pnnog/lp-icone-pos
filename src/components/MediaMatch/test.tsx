import { render, screen } from '@testing-library/react';

import MediaMatch from '.';

describe('<MediaMatch />', () => {
  it('should render MediaMatch', () => {
    render(<MediaMatch greaterThan="medium"> Não deve aparecer </MediaMatch>);
    expect(screen.getByText(/Não deve aparecer/i)).toHaveStyleRule(
      'display',
      'none'
    );
  });

  it('doest not should render MediaMatch', () => {
    const { container } = render(
      <MediaMatch greaterThan="medium"> Deve aparecer </MediaMatch>
    );
    expect(screen.getByText(/Deve aparecer/i)).toHaveStyleRule(
      'display',
      'block',
      {
        media: '(min-width:768px)'
      }
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
