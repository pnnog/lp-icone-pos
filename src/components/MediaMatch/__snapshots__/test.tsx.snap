// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaMatch /> doest not should render MediaMatch 1`] = `
.c0 {
  display: none;
}

@media (min-width:768px) {
  .c0 {
    display: block;
  }
}

<div
  class="c0"
>
   Deve aparecer 
</div>
`;
